packages/appeals-service-api/src/routes/v2/appeal-cases/repo.js

```diff
@@ -39,17 +82,142 @@ class AppealCaseRepository {
 	 * Get an appeal by case reference (aka appeal number)
 	 *
 	 * @param {string} caseReference
-	 * @param {AppealCaseCreateInput} data
+	 * @param {AppealHASCase} data
 	 * @returns {Promise<AppealCase>}
 	 */
-	putByCaseReference(caseReference, data) {
-		return this.dbClient.appealCase.upsert({
-			create: { ...data, Appeal: { create: {} } }, // create with parent Appeal
-			update: data,
+	async putHASByCaseReference(caseReference, data) {
+		const mappedData = mapHASDataModelToAppealCase(data);
+
+		// todo: move this to mapping
+		delete mappedData.caseType;
+		delete mappedData.linkedCaseStatus;
+		delete mappedData.leadCaseReference;
+		delete mappedData.notificationMethod;
+		delete mappedData.nearbyCaseReferences;
+		delete mappedData.neighbouringSiteAddresses;
+		delete mappedData.affectedListedBuildingNumbers;
+
+		const appealCase = await this.dbClient.appealCase.upsert({
+			create: { ...mappedData, Appeal: { create: {} } },
+			update: mappedData,
 			where: {
 				caseReference
 			}
 		});
+
+		// case relations
+		await this.dbClient.$transaction(async (tx) => {
```

Comment:
seemed to have issues within one transaction with timeouts so instead have a transaction for each many-many
Could increase timeout and do in one transaction, but think it seems ok to get the case in first.

Also can look at Promise.all here

Reply:
I've gone with dumping JSON into some nvarchar max fields, think it should be fine and can always migrate to a new field if needed

Reply:
looks like you can configure a timeout if you get this issue again https://www.prisma.io/docs/orm/prisma-client/queries/transactions#:~:text=//%20default%3A%202000-,timeout%3A%2010000%2C%20//%20default%3A%205000,-isolationLevel%3A%20Prisma

Reply:
sorry @beejjacobs what are you suggesting here?

Reply:
I've figured it out